src/Compilers/Core/Portable/SourceGenerator/SourceGeneratorExtensions.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Collections;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public static class SourceGeneratorExtensions
+    {
+        public static ImmutableArray<SyntaxTree> GenerateSource(
+            this Compilation compilation,
+            ImmutableArray<SourceGenerator> generators,
+            string path,
+            bool writeToDisk,
+            CancellationToken cancellationToken)
+        {
+            var builder = ArrayBuilder<SyntaxTree>.GetInstance();
```

Comment:
Yes, this could be sized to `generators.Length` although that assumes that each generator typically generates one source file. Will leave as is for now.